client/src/main/java/io/split/client/SplitClientImpl.java

```diff
@@ -341,6 +371,86 @@ private Map<String, SplitResult> getTreatmentsWithConfigInternal(String matching
         }
     }
 
+    private Map<String, SplitResult> getTreatmentsWithSetsAndWithConfigInternal(String matchingKey, String bucketingKey, List<String> sets,
+                                                                     Map<String, Object> attributes, MethodEnum methodEnum) {
+        long initTime = System.currentTimeMillis();
+        if (sets == null || sets.isEmpty()) {
+            _log.warn(String.format("%s: sets must be a non-empty array", methodEnum.getMethod()));
+            return new HashMap<>();
+        }
+        FlagSetsValidResult flagSetsValidResult = FlagSetsValidator.areValid(sets);
+        try {
+            if (!flagSetsValidResult.getValid()) {
+                _log.warn("The sets are invalid");
+                return new HashMap<>();
+            }
+            if (filterSetsAreInConfig(flagSetsValidResult.getFlagSets()).isEmpty()) {
+                _log.warn("The sets are not");
```

Comment:
Something seems to be missing from the message

Reply:
Updated.